### ISSUE TITLE
double free or corruption (!prev) in Docker image tensorflow/tensorflow:1.0.0

### ISSUE THREAD

**Issue by lienhua34** [tensorflow/tensorflow#7960]
NOTE: Only file GitHub issues for bugs and feature requests.  All other topics will be closed.

For general support from the community, see [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).
To make bugs and feature requests more easy to find and organize, we close issues that are deemed
out of scope for GitHub Issues and point people to StackOverflow.

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.

### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?

https://github.com/tensorflow/tensorflow/issues/6968
https://github.com/tensorflow/tensorflow/issues/7839

### Environment info

I run tensorflow in Docker image [tensorflow/tensorflow:1.0.0](https://hub.docker.com/r/tensorflow/tensorflow/).

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

I used tensorflow/tensorflow:1.0.0 to run [examples/tutorials/mnist/mnist_softmax.py](https://github.com/tensorflow/tensorflow/blob/r1.0/tensorflow/examples/tutorials/mnist/mnist_softmax.py). My docker run command is:

```shell
$ docker run -ti \
    -v /Users/lienhua34/Programs/python/tensorflow/tensorflow/examples/tutorials/mnist:/mnist \
   tensorflow/tensorflow:1.0.0 /bin/bash
```

Then cd to /mnist, and run mnist_softmax.py:

```shell
root@5a17d6873afc:/notebooks# cd /mnist/
root@5a17d6873afc:/mnist# ls -l
total 52
-rw-r--r-- 1 root root 2581 Feb 16 15:42 BUILD
-rw-r--r-- 1 root root  979 Feb 16 15:42 __init__.py
drwxr-xr-x 6 root root  204 Mar  1 03:41 data
-rw-r--r-- 1 root root 9515 Feb 16 15:42 fully_connected_feed.py
-rw-r--r-- 1 root root 1107 Feb 16 15:42 input_data.py
-rw-r--r-- 1 root root 5292 Feb 16 15:42 mnist.py
-rw-r--r-- 1 root root 2772 Feb 17 09:01 mnist_softmax.py
-rw-r--r-- 1 root root 3684 Feb 16 15:42 mnist_softmax_xla.py
-rw-r--r-- 1 root root 8391 Feb 16 15:42 mnist_with_summaries.py
root@5a17d6873afc:/mnist# python mnist_softmax.py --data_dir=data
Loading data....
Extracting data/train-images-idx3-ubyte.gz
Extracting data/train-labels-idx1-ubyte.gz
Extracting data/t10k-images-idx3-ubyte.gz
Extracting data/t10k-labels-idx1-ubyte.gz
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE3 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
0.919
*** Error in `python': double free or corruption (!prev): 0x000000000179b6c0 ***
Aborted
```

### What other attempted solutions have you tried?

I followed the issue https://github.com/tensorflow/tensorflow/issues/6968 to install libtcmalloc-minimal4,

```shell
sudo apt-get install libtcmalloc-minimal4
export LD_PRELOAD="/usr/lib/libtcmalloc_minimal.so.4"
```

After that, it will run fine.

```
root@5b7be5bcf590:/mnist# python mnist_softmax.py --data_dir=data
Loading data....
Extracting data/train-images-idx3-ubyte.gz
Extracting data/train-labels-idx1-ubyte.gz
Extracting data/t10k-images-idx3-ubyte.gz
Extracting data/t10k-labels-idx1-ubyte.gz
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE3 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
0.9212
```

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

**Comment by cancan101** [comment]
Upgrade to the one of the docker images based of 16.04.

**Comment by lienhua34** [comment]
@cancan101 Which one? https://hub.docker.com/r/tensorflow/tensorflow/tags/

**Comment by cancan101** [comment]
I use `1.0.0.2-devel-gpu` and it is based on 16.04.

**Comment by lienhua34** [comment]
But I don't want to use GPU.

**Comment by cancan101** [comment]
Then grab another image with the same version number

**Comment by lienhua34** [comment]
@cancan101 Sorry, I don't understand.

**Comment by gunan** [comment]
If you do not need gpu, you can simply use the `1.0.0.2-devel` docker image. Same version as the one @cancan101 suggested, but no gpu.

**Comment by gunan** [comment]
Closing this issue, as it sounds like it was resolved post 1.0.
